### PR TITLE
cardano-testnet: Increase Lovelace supply, Fix cleanup logic, Some refactoring

### PR DESCRIPTION
**Pull request by errfrom**
https://github.com/IntersectMBO/cardano-node/issues/5953

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior are covered with tests
- [x] The template (`templates/ctl-scaffold`) has been [updated](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/development.md#maintaining-the-template)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Changed`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: src/Internal/Testnet/Contract.purs [hunk 1]
```diff
@@ -304,6 +305,13 @@ execDistrFundsPlan withCardanoCliUtxos rounds = do
     )
     roundsFixed
 
+newtype KeyWalletShow = KeyWalletShow KeyWallet
```
Comment (klntsky) [inline src/Internal/Testnet/Contract.purs:308]: What's the purpose?

Comment (klntsky) [reply src/Internal/Testnet/Contract.purs:308]: Where is it used?

File: src/Internal/Testnet/DistributeFunds.purs [hunk 2]
```diff
@@ -1,252 +1,273 @@
...
-assignUtxo _ _ Nil = Left
-  { err: "Ran out of sources", acc: Nil }
-assignUtxo thresholds utxo acc@(Cons source sources)
-  | 0 >= thresholds.maxTargetUtxosPerTx =
-      Left { err: "maxTargetUtxosPerTx must be greater than 1", acc }
-  | utxo.amount >= thresholds.maxCoinPerTx =
-      Left
-        { err: "UTxO required amount is higher than the maxCoinPerTx threshold"
-        , acc
+  => DistrFundsParams wallet amount
```
Comment (klntsky) [inline src/Internal/Testnet/DistributeFunds.purs:148]: A written description would help

File: src/Internal/Testnet/DistributeFunds.purs [hunk 3]
```diff
@@ -1,252 +1,273 @@
...
+  { sources :: List (SourceState wallet amount)
+  , deferredTargets :: List (wallet /\ amount)
   }
 
-parallelizedDistributionPlan
-  :: forall src target amount
-   . Map src (Array (Tx Unit target amount))
-  -> Array (Map src (Tx Unit target amount))
-parallelizedDistributionPlan _ = unsafeThrow "hello"
+data AssignUtxoResult wallet amount
```
Comment (klntsky) [inline src/Internal/Testnet/DistributeFunds.purs:133]: Please explain the meaning of each alternative

File: src/Internal/Testnet/DistributeFunds.purs (hunk 2 above)
Comment (errfrom) [reply src/Internal/Testnet/DistributeFunds.purs:186]: Addressed in https://github.com/Plutonomicon/cardano-transaction-lib/pull/1644/commits/b881710f7c5cbda8533a663d9caf5af252fc6329

File: src/Internal/Testnet/DistributeFunds.purs (hunk 3 above)
Comment (errfrom) [reply src/Internal/Testnet/DistributeFunds.purs]: Addressed in https://github.com/Plutonomicon/cardano-transaction-lib/pull/1644/commits/b881710f7c5cbda8533a663d9caf5af252fc6329

File: src/Internal/Testnet/Contract.purs (hunk 1 above)
Comment (errfrom) [reply src/Internal/Testnet/Contract.purs]: This is a wrapper for KeyWallet with a Show instance, which is needed for better DistrFundsError messages

File: src/Internal/Testnet/Server.purs [hunk 4]
```diff
@@ -281,129 +277,128 @@ startTestnetCluster cfg cleanupRef = do
...
   env <- liftEffect Node.Process.getEnv
-  -- initCwd <- liftMaybe (error "Couldn't find INIT_CWD env variable")
-  --   $ Object.lookup "INIT_CWD" env
   let
     env' = Object.fromFoldable
-      [ "TMPDIR" /\ cwd -- only for 8.1.1; 8.7.2 puts it's testnet directory into cwd instead
-      -- , "CARDANO_NODE_SRC" /\ (initCwd <</>> "cardano-testnet-files")
-      , "CARDANO_CLI" /\ "cardano-cli"
-      , "CREATE_SCRIPT_CONTEXT" /\ "create-script-context"
+      [ "CARDANO_CLI" /\ "cardano-cli"
```
Comment (klntsky) [inline src/Internal/Testnet/Server.purs:286]: Is it needed?

**Review by klntsky — APPROVED** [review]

File: src/Internal/Testnet/Server.purs (hunk 4 above)
Comment (errfrom) [reply src/Internal/Testnet/Server.purs:286]: Yes, without those vars, cardano-testnet will fail to run cardano-node and cardano-cli

**Pull request merged**